### PR TITLE
Added paranthesis checker function in python

### PR DESCRIPTION
**Pull request by Shishir198**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

YES

### Description

The function can support all the common,curly,square and angle brackets in program and detect whether Brackets are balanced or not, using hashmap in O(n) complexity

### Checklist

- [y ] I've read the contribution guidelines.
- [y ] I've checked the issue list before deciding what to submit.
- [ y] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

(Write your answer here.)

### CODE REVIEW

File: Python/other/paranthesis_checker.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
+def checkParanthesis(s):
+    # hashmap maintaing the open and closed brackets
+    occurence = {"(":0 , ")":0, "{":0 , "}":0 , "[":0 , "]" : 0 , "<":0 , ">":0}
+    # occurence supports common,curly,square and angle bracket
+
+    for i in s:
```
Comment (SKAUL05) [inline Python/other/paranthesis_checker.py]: Use meaningful names instead of alphabets

File: Python/other/paranthesis_checker.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
+def checkParanthesis(s):
```
Comment (SKAUL05) [inline Python/other/paranthesis_checker.py]: Use Docstrings for documentation

File: Python/other/paranthesis_checker.py [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+        if occurence["["] == occurence["]"] and occurence["<"] == occurence[">"]:
+            print("Yes")
+        else:
+            print("No")
+    else:
+        print("No")
+
+
+if __name__ == "__main__":
+    inp = input()
```
Comment (SKAUL05) [inline Python/other/paranthesis_checker.py]: Ask user for input.

**Review by SKAUL05 — CHANGES_REQUESTED** [review]

**Comment by atarax665** [comment]
Update README.md

**Comment by Shishir198** [comment]
@atarax665 README.md updated and pushed

**Comment by Shishir198** [comment]
@SKAUL05 Hello sir,as per your request I have incorporated the changes.

**Review by siddharth25pandey — CHANGES_REQUESTED** [review]
@Shishir198  Remove the merge Conflicts Error

File: Python/other/paranthesis_checker.py [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    # hashmap maintaing the open and closed brackets
+    occurence = {"(":0 , ")":0, "{":0 , "}":0 , "[":0 , "]" : 0 , "<":0 , ">":0}
+    # occurence supports common,curly,square and angle bracket
+
+    for bracket in input_string:
+        occurence[bracket]+=1
+
+    if occurence["("] == occurence[")"] and occurence["{"] == occurence["}"]:
+        if occurence["["] == occurence["]"] and occurence["<"] == occurence[">"]:
+            print("Yes")
```
Comment (atarax665) [inline Python/other/paranthesis_checker.py]: It would be better to return Balanced/Not balanced instead of Yes/No.

**Comment by Shishir198** [comment]
@siddharth25pandey  I have resolved the conflicts .... @atarax665 I have made the changes and pushed.

**Comment by Shishir198** [comment]
@siddharth25pandey @atarax665 Why I am always getting merge conflicts . Kindly help.

File: Python/other/paranthesis_checker.py [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+        else:
+            print("Not Balanced")
+    else:
+        print("Not Balanced")
+
+
+if __name__ == "__main__":
+    inp = input("Enter the Brackets Combinations\n")
+    checkParanthesis(inp)
+
```
Comment (HarshCasper) [inline Python/other/paranthesis_checker.py:30]: Add sample Input/Output for the Code

File: Python/other/paranthesis_checker.py [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+    ''' 
+        argument : string
+
+        Function checks whether the Brackets Combinations Provided As Input 
+        is Balanced or Not.
+        
+        usage : checkParanthesis(string)
+        
+    '''
+
```
Comment (HarshCasper) [inline Python/other/paranthesis_checker.py:8]: Remove these blank lines

**Review by HarshCasper — CHANGES_REQUESTED** [review]

File: Python/other/paranthesis_checker.py (hunk 6 above)
Comment (Shishir198) [reply Python/other/paranthesis_checker.py:8]: @HarshCasper I have completed your desired changes .Kindly look through it.

File: Python/other/paranthesis_checker.py [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+    else:
+        print("Not Balanced")
+
+
+if __name__ == "__main__":
+    inp = input("Enter the Brackets Combinations\n")
+    checkParanthesis(inp)
+    
+
+    # Sample output 1:
```
Comment (atarax665) [inline Python/other/paranthesis_checker.py:32]: Use multi-line comments for sample input/output. Also add time and space complexity for your program.

**Review by atarax665 — COMMENTED** [review]
Make sure you code follows PEP8 guidelines. You can check it [here](http://pep8online.com/).

**Comment by HarshCasper** [comment]
I'm closing the PR due to no activity. If you would like to reopen this PR to be reviewed please join the [Slack](https://join.slack.com/t/tesseractcodi-bxq5968/shared_invite/zt-ju6c0nqq-KgkTvFyxjNVbDRBIjADanw) and drop me a message.

**Pull request closed without merging**